appsmithorg/appsmith
# fix: specific sheets to whole drive issue fixed

## Description

This PR fixes the issue:
- Spreadsheet dropdown does not show all sheets, when google sheet datasource is edited from specific sheets to all sheets.
- If files are picked in the file picker, no error message is shown on datasource review page, but if files are not picked, error message is shown on review page.

Fixes #21916 

### How to test
First Issue:
- Create google sheet datasource with specific sheets modality
- Now create a query and check that only specific sheets are visible in spreadsheet dropdown
- Now edit the datasource and change scope from specific to all sheets
- Check the query and spreadsheet dropdown should show all sheets, as scope has changed

Second Issue:
- Create google sheet datasource, select files from file picker, no error message should be seen
- Edit the datasource, do not select files from file picker, error message should be seen.


## Type of change
- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

- Manual

### Test Plan
> Add Testsmith test cases links that relate to this PR

### Issues raised during DP testing
> Link issues raised during DP testing for better visiblity and tracking (copy link from comments dropped on this PR)


## Checklist:
### Dev activity


### QA activity:
